crates/tests/src/e2e/ibc_tests.rs

```diff
@@ -2739,112 +1551,17 @@ fn check_gaia_balance(
     Ok(())
 }
 
-/// Check balances after non IBC transfer
-fn check_balances_after_non_ibc(
-    port_id: &PortId,
-    channel_id: &ChannelId,
-    test_b: &Test,
-) -> Result<()> {
-    // Check the source on Chain B
-    let ibc_denom = format!("{port_id}/{channel_id}/nam");
-    check_balance(test_b, BERTHA, &ibc_denom, 50_000_000_000)?;
-
-    // Check the traget on Chain B
-    check_balance(test_b, ALBERT, &ibc_denom, 50_000_000_000)?;
-
-    Ok(())
-}
-
-/// Check balances after IBC transfer back
-fn check_balances_after_back(
-    dest_port_id: &PortId,
-    dest_channel_id: &ChannelId,
-    test_a: &Test,
-    test_b: &Test,
-) -> Result<()> {
-    // Check the escrowed balance on Chain A
-    let escrow = Address::Internal(InternalAddress::Ibc).to_string();
-    check_balance(test_a, escrow, NAM, 50000)?;
-    // Check the source balance on Chain A
-    check_balance(test_a, ALBERT, NAM, 1950000)?;
-
-    // Check the balance on Chain B
-    let ibc_denom = format!("{dest_port_id}/{dest_channel_id}/nam");
-    check_balance(test_b, BERTHA, ibc_denom, 0)?;
-
-    Ok(())
-}
-
-/// Check balances after IBC shielded transfer
-fn check_shielded_balances(
-    dest_port_id: &PortId,
-    dest_channel_id: &ChannelId,
-    test_a: &Test,
-    test_b: &Test,
-) -> Result<()> {
-    // Check the shielded balance on Chain A
-    check_balance(test_a, AA_VIEWING_KEY, BTC, 90)?;
-
-    // Check the shielded balance on Chain B
-    let ibc_denom = format!("{dest_port_id}/{dest_channel_id}/btc");
-    check_balance(test_b, AB_VIEWING_KEY, ibc_denom, 1_000_000_000)?;
-
-    Ok(())
-}
-
-fn check_funded_balances(
-    dest_port_id: &PortId,
-    dest_channel_id: &ChannelId,
-    test_b: &Test,
+fn check_inflated_balance(
+    test: &Test,
```

Comment:
Shouldn't we pass here also the initial balance of the viewing key? In this function I see that we check:

`assert!(balance > 0.0);`

but I'm not sure this is completely correct: if we want to check the shielded rewards it means the viewing key will have some balance to produce the rewards. Checking just for the balance to be positive doesn't guarantee we have distributed rewards (or that the specified target has received them): I believe we should pass in the balance from the previous masp epoch and then check that the new balance is greater than this one

Reply:
The account's NAM balance should be empty before the inflation.
I added the check. Thanks.